### ISSUE TITLE
Windows 7 fails at Completing setup of Qubes Windows Tools

### ISSUE THREAD

**Issue by NipponBill** [ElliotKillick/qvm-create-windows-qube#78]
**TLDR;**
Attempting to install Windows 7 and the script fails at

- **Completing setup of Qubes Windows Tools...**

from `dom0` I run:
`qvm-create-windows-qube -n sys-firewall -soy -i win7x64sp1.iso -a win7x64-ultimate.xml win7`

Everything runs normally until after the script gets through **Installing Qubes Windows Tools...**

Windows opens up to a Startup Repair dialog. It then restarts the VM to complete the repairs. Then the `dom0` script does nothing. It seems to hang.
```
[user@dom0 Desktop]$ qvm-create-windows-qube -n sys-firewall -soy -i win7sp1.iso -a win7x64-ultimate.xml win7
[i] Preparing Windows media for automatic installation...
[i] Creating loop device from ISO...
[i] Mounting loop device...
[i] Creating new ISO...
Booting catalog starts at sector: 22 
Manufacturer of CD: Microsoft Corporation
Image architecture: x86
Boot media type is: no emulation
El Torito image starts at sector 734 and has 8 sector(s) of 512 Bytes

Image has been written to file "/tmp/boot.bin.aCqzb".
[i] Unmounting loop device...
Unmounted /dev/loop2.
[i] Deleting loop device...
[i] Deleting temporary boot image...
[+] Created automatic Windows installation media for win7sp1.iso successfully!
[i] Starting creation of win7
[i] Starting first part of Windows installation process...
[i] Starting second part of Windows installation process...
[i] Preparing Qubes Windows Tools for automatic installation...
[i] Installing Qubes Windows Tools...
[i] Completing setup of Qubes Windows Tools...
```
I have the [tabit-pro QWT](https://github.com/tabit-pro/qubes-windows-tools-cross/releases/tag/v4.1.67) installed in `dom0`. This is a 64bit Win7 ISO with SP1. I'm uncertain how to get beyond this step.

**Comment by d2x** [comment]
While I haven't found the code yet to verify, I believe this may be because QWT attempts to download the drivers from the xen website and they have been removed. Only version 9.0.0 remains ([here](https://xenbits.xenproject.org/pvdrivers/win/)).

**Comment by d2x** [comment]
I have the same issue on Windows 10 Pro with QWT I compiled myself using the 9.0.0 PV drivers. I get stuck at the line prior with the Windows machine powered off.

```
$ qvm-create-windows-qube -i Win10_x64.iso -a win10x64-pro.xml -toy -n sys-firewall win-10-template
[i] Preparing Windows media for automatic installation...
[i] Starting creation of win-10-template
[i] Starting first part of Windows installation process...
[i] Starting second part of Windows installation process...
[i] Preparing Qubes Windows Tools for automatic installation...
```

It looks like the auto-qwt was built to not pull in the QWT installer package from the ISO due to potential security risks.

`installer/README.txt`:

```
Qubes Windows Tools are currently unavailable due to security concerns.
See https://github.com/QubesOS/qubes-secpack/blob/master/QSBs/qsb-091-2023.txt for details.
```

As Windows is properly installed, you can allow unsigned drivers, reboot and run the QWT installer manually. This is only advised with an updated QWT package that utilizes 9.0.0.

**Comment by d2x** [comment]
> It looks like the auto-qwt was built to not pull in the QWT installer package from the ISO due to potential security risks.
> 
> `installer/README.txt`:
> 
> ```
> Qubes Windows Tools are currently unavailable due to security concerns.
> See https://github.com/QubesOS/qubes-secpack/blob/master/QSBs/qsb-091-2023.txt for details.
> ```

I imagine this part was because I didn't delete my `windows-mgmt` AppVM and run the installer again. I think the QWT ISO is copied over from dmo0 during the `windows-mgmt` AppVM creation.

**Comment by NipponBill** [comment]
@d2x Where are you getting the updated QWT package that utilizes 9.0.0?

**Comment by d2x** [comment]
> @d2x Where are you getting the updated QWT package that utilizes 9.0.0?

I built it myself. I posted high level overview [here in the forums](https://forum.qubes-os.org/t/windows-tools-for-4-2/22708/12).

**Comment by ElliotKillick** [comment]
Use the Windows 7 ISO download provided by Mido. It includes an MSU that adds support for SHA-256 driver signatures (Xen drivers only have SHA-256 signatures with no SHA-1). This is a known issue documented in the README, which only applies to Windows 7.

**Comment by ElliotKillick** [comment]
@d2x The answer files start executing from run.bat, this is the correct entry point to begin the QWT installation process: https://github.com/ElliotKillick/qvm-create-windows-qube/blob/master/tools/auto-qwt/run.batrun.bat runs trust-certificates.bat to add Xen certificates, so these prompts don't show up. However, these may not work now that people have to use unsigned or self-built drivers. In this case, allow-drivers.vbs should work as a substitute. allow-drivers.vbs is a hack originally created to accept this prompt on Win7/2008R2, where MS intentionally broke the trust-certificates.bat method. However, now it's also being used to automate accepting the prompt on newer Windows versions: https://github.com/ElliotKillick/qvm-create-windows-qube/blob/master/tools/auto-qwt/allow-drivers.vbs

**Comment by jpph** [comment]
I had same issue with win10. 
If it can take other I did whats is said in the last message of this thread :

https://forum.qubes-os.org/t/qvm-create-windows-qube-qubes-windows-tools-install-hangs-infinitely/9851/17

(basically you need ot update the iso of the windows-tool in dom0)

**Comment by jmynes** [comment]
I am also getting the same issue with the `mido.sh` provided win 7 x64 ultimate iso
The startup repair message is `Boot critical file: c:\windows\system32\drivers\xenbus.sys is corrupt.`:

<img width="808" height="628" alt="Image" src="https://github.com/user-attachments/assets/cc848211-6e9f-4fac-8850-e69386a5b237" />

**Comment by jmynes** [comment]
I am now able to reliably set up a Windows 7 x64 Ultimate qube in QubesOS 4.2.4 using `qubes-windows-tools-4.1.69-1.fc37.noarch.rpm` and a specific order of commands I have in a few bash scripts that wrap this repo.

I can try to sift through them and figure out what exactly I am doing differently, if someone wants Windows 7 to work on their end.

I have not tested on QubesOS 4.3 or any rc
